normandy/recipes/signing.py

```diff
@@ -190,9 +193,26 @@ def verify_x5u(url, expire_early=None):
     chain. Otherwise, raise an exception explaining why they are not
     valid.
     """
-    req = requests.get(url)
-    req.raise_for_status()
-    pem = req.content.decode()
+    cache_key = f"fetch-x5u-pem::{url}"
+
+    pem = None
+    if settings.X5U_CACHE_TIME:
+        pem = cache.get(cache_key)
+
+    if isinstance(pem, Exception):
+        raise CachedRequestException(url=url)
+
+    if pem is None:
+        try:
+            resp = requests.get(url, timeout=X5U_REQUEST_TIMEOUT)
+            resp.raise_for_status()
+        except requests.RequestException as exc:
+            cache.set(cache_key, exc, settings.X5U_CACHE_TIME)
+            raise
+
+        pem = resp.content.decode()
+        if settings.X5U_CACHE_TIME:
```

Comment:
This is to guard against bothering with caching when TTL is 0? If that is helpful here, would it also be helpful at line 210?

Reply:
This was specifically so that tests could totally disable caching (we can't cache mock objects), but yes, it would be helpful on line 210 as well.